server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/AddUserToTemplateAction.java

```diff
@@ -108,21 +104,8 @@ private static AddUserToTemplateWsRequest toAddUserToTemplateWsRequest(Request r
   }
 
   private boolean isUserAlreadyAdded(DbSession dbSession, long templateId, String userLogin, String permission) {
-    OldPermissionQuery permissionQuery = OldPermissionQuery.builder().permission(permission).membership(IN).build();
-    List<UserWithPermissionDto> usersWithPermission = dbClient.permissionTemplateDao().selectUsers(dbSession, permissionQuery, templateId, 0, Integer.MAX_VALUE);
-    return from(usersWithPermission).anyMatch(new HasUserPredicate(userLogin));
-  }
-
-  private static class HasUserPredicate implements Predicate<UserWithPermissionDto> {
-    private final String userLogin;
-
-    public HasUserPredicate(String userLogin) {
-      this.userLogin = userLogin;
-    }
-
-    @Override
-    public boolean apply(UserWithPermissionDto userWithPermission) {
-      return userLogin.equals(userWithPermission.getLogin());
-    }
+    PermissionQuery permissionQuery = PermissionQuery.builder().setPermission(permission).build();
+    List<String> usersWithPermission = dbClient.permissionTemplateDao().selectUserLoginsByQueryAndTemplate(dbSession, permissionQuery, templateId);
+    return usersWithPermission.stream().anyMatch(s -> s.equals(userLogin));
```

Comment:
not sure to having one long line would help. I prefer having a intermediary variable with a self-descriptive name.